corehq/apps/locations/permissions.py

```diff
@@ -283,8 +283,16 @@ def conditionally_location_safe(conditional_function):
 
     """
     def _inner(view_fn):
-        if isinstance(view_fn, type) and issubclass(view_fn, GenericReportView):
-            CONDITIONALLY_LOCATION_SAFE_HQ_REPORTS[view_fn.slug] = conditional_function
+        if isinstance(view_fn, type):
+
+            # Django class-based views
+            if issubclass(view_fn, View):
+                view_fn.dispatch.__func__._conditionally_location_safe_function = conditional_function
```

Comment:
Looks like it'll need to drop the `__func__` part, yeah?  I can play with it and see if something like this would work:
```python
def set_method_attr(method, attr, value):
    if six.PY3:
        setattr(method, attr, value)
    else:
        setattr(method.__func__, attr, value)
```